sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/ConfigurationUtils.java

```diff
@@ -0,0 +1,121 @@
+     *
+     * @param name property name.
+     * @return instance of {@link ConfigurationPropertyBuilder}.
+     */
+    public static ConfigurationPropertyBuilder<Integer> integerSharedPropertyBuilder(String name) {
+        return new ConfigurationPropertyBuilder<>(name, INTEGER_CONVERTER).canLogValue(true).shared(true);
+    }
+
+    /**
+     * Creates {@link ConfigurationPropertyBuilder} configured to log property value and
+     * parses value as long number of milliseconds, proxying  {@link NumberFormatException} exception.
+     *
+     * @param name property name.
+     * @return instance of {@link ConfigurationPropertyBuilder}.
+     */
+    public static ConfigurationPropertyBuilder<Duration> durationSharedPropertyBuilder(String name) {
+        return new ConfigurationPropertyBuilder<>(name, DURATION_CONVERTER).canLogValue(true).shared(true);
+    }
+
+    /**
+     * Creates {@link ConfigurationPropertyBuilder} configured to log property value and
+     * parse value using {@link Boolean#parseBoolean(String)}.
+     *
+     * @param name property name.
+     * @return instance of {@link ConfigurationPropertyBuilder}.
+     */
+    public static ConfigurationPropertyBuilder<Boolean> booleanSharedPropertyBuilder(String name) {
+        return new ConfigurationPropertyBuilder<>(name, BOOLEAN_CONVERTER).canLogValue(true).shared(true);
+    }
+
+    /**
+     * Attempts to convert the configuration value to given primitive {@code T} using
+     * corresponding {@code parse} method on this type.
+     *
+     * <p><b>Following types are supported:</b></p>
+     * <ul>
+     * <li>{@link Byte}</li>
+     * <li>{@link Short}</li>
+     * <li>{@link Integer}</li>
+     * <li>{@link Long}</li>
+     * <li>{@link Float}</li>
+     * <li>{@link Double}</li>
+     * <li>{@link Boolean}</li>
+     * </ul>
+     *
+     * If the value is null or empty then the default value is returned.
+     *
+     * @param value Configuration value retrieved from the map.
+     * @param defaultValue Default value to return if the configuration value is null or empty.
+     * @param <T> Generic type that the value is converted to if not null or empty.
+     * @return The converted configuration, if null or empty the default value.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T convertToPrimitiveOrDefault(String value, T defaultValue) {
+        // Value is null or empty, return the default.
+        if (CoreUtils.isNullOrEmpty(value)) {
+            return defaultValue;
+        }
+
+        // Check the default value's type to determine how it needs to be converted.
+        Object convertedValue;
+        if (defaultValue instanceof Byte) {
+            convertedValue = Byte.parseByte(value);
+        } else if (defaultValue instanceof Short) {
+            convertedValue = Short.parseShort(value);
+        } else if (defaultValue instanceof Integer) {
```

Comment:
If I pass an `int` value does `instanceof Integer` return true?

Reply:
Yes, it does.